test/e2e/kubevirt.go

```diff
@@ -870,160 +876,402 @@ passwd:
+						},
+					},
+				},
+			}
+			if td.mode == kubevirtv1.MigrationPostCopy {
+				err = crClient.Create(context.TODO(), forcePostCopyMigrationPolicy)
 				Expect(err).ToNot(HaveOccurred())
-				return vm.Status.Ready
-			}).WithPolling(time.Second).WithTimeout(5 * time.Minute).Should(BeTrue())
+				defer func() {
+					Expect(crClient.Delete(context.TODO(), forcePostCopyMigrationPolicy)).To(Succeed())
+				}()
+			}
+
+			vmLabels := map[string]string{}
+			if td.mode == kubevirtv1.MigrationPostCopy {
+				vmLabels = forcePostCopyMigrationPolicy.Spec.Selectors.VirtualMachineInstanceSelector
+			}
+			vms, err := composeDefaultNetworkLiveMigratableVMs(td.numberOfVMs, vmLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, vm := range vms {
+				By(fmt.Sprintf("Create virtual machine %s", vm.Name))
+				vmCreationRetries := 0
+				Eventually(func() error {
+					if vmCreationRetries > 0 {
+						// retry due to unknown issue where kubevirt webhook gets stuck reading the request body
+						// https://github.com/ovn-org/ovn-kubernetes/issues/3902#issuecomment-1750257559
+						By(fmt.Sprintf("Retrying vm %s creation", vm.Name))
+					}
+					err = crClient.Create(context.Background(), vm)
+					vmCreationRetries++
+					return err
+				}).WithPolling(time.Second).WithTimeout(time.Minute).Should(Succeed())
+			}
+
+			if td.shouldExpectFailure {
+				By("annotating the VMI with `fail fast`")
+				vmKey := types.NamespacedName{Namespace: namespace, Name: "worker1"}
+				var vmi kubevirtv1.VirtualMachineInstance
+				Eventually(func() error {
+					return crClient.Get(context.TODO(), vmKey, &vmi)
+				}).WithPolling(time.Second).WithTimeout(time.Minute).Should(Succeed())
+
+				vmi.ObjectMeta.Annotations[kubevirtv1.FuncTestLauncherFailFastAnnotation] = "true"
+
+				Expect(crClient.Update(context.TODO(), &vmi)).To(Succeed())
+			}
+
+			for _, vm := range vms {
+				By(fmt.Sprintf("Waiting for readiness at virtual machine %s", vm.Name))
+				Eventually(func() bool {
+					err = crClient.Get(context.Background(), crclient.ObjectKeyFromObject(vm), vm)
+					Expect(err).ToNot(HaveOccurred())
+					return vm.Status.Ready
+				}).WithPolling(time.Second).WithTimeout(5 * time.Minute).Should(BeTrue())
+			}
+			wg.Add(int(td.numberOfVMs))
+			for _, vm := range vms {
+				go runLiveMigrationTest(td, vm)
+			}
+			wg.Wait()
+		},
+			Entry("with pre-copy succeeds, should keep connectivity", liveMigrationTestData{
+				mode:        kubevirtv1.MigrationPreCopy,
+				numberOfVMs: 1,
+			}),
+			Entry("with post-copy succeeds, should keep connectivity", liveMigrationTestData{
+				mode:        kubevirtv1.MigrationPostCopy,
+				numberOfVMs: 1,
+			}),
+			Entry("with pre-copy fails, should keep connectivity", liveMigrationTestData{
+				mode:                kubevirtv1.MigrationPreCopy,
+				numberOfVMs:         1,
+				shouldExpectFailure: true,
+			}),
+		)
+	})
+	Context("with non default network and persistent ips configured", func() {
```

Comment:
is mthis more rather `with secondary network...`, like the default network is still there and is also live migrated as in the previous spec?

Reply:
this is a good point: we could pin-point the feature and just drop the cluster default network from the VM. i.e. only have a single secondary network interface.

Reply:
I will reword it as "with secondary network"

Reply:
and how about dropping the default cluster network attachment from the VM's interface list ? 

That has nothing to do with the feature.
@qinqon

Reply:
My bad then. All good.